content/docs/user-guide/visualization/environments/terrain/terrain-developer-guide/system-extensions.md

```diff
@@ -0,0 +1,48 @@
+
+Ideally, new gradient components would also support the following:
+
+* Use the **Shape** components to define how the gradient data should map into world space.
+* Support the **Gradient Transform Modifier** component for performing arbitrary transforms to the gradient data.
+* By convention, all gradients currently listen to `OnEntityVisibilityChanged` and use that to enable or disable the gradient data in the Editor. This makes it easy to turn gradients on and off while authoring, though it's important to recognize that they will still be active at runtime regardless of the authoring visibility setting.
+
+## Terrain Components
+
+Each of the terrain components can be replaced with new components that meet the same requirements. For example, a new component could be created that streams down real-world satellite height data as a replacement for the **Terrain Height Gradient List**. It's also possible to combine components together, such as creating a simplified Terrain Spawner component that implements the functionality of the **Terrain Layer Spawner**, **Terrain Height Gradient List**, and **Terrain Surface Gradient List** on one single component.
+
+Any replacement components need to support the following:
+
+* Implement the request bus that matches the component being replaced (`TerrainSpawnerRequestBus`, `TerrainAreaHeightRequestBus`, `TerrainAreaSurfaceRequestBus`, etc).
+* Declare that it implements each service that matches the component being replaced (`TerrainAreaService`, `TerrainHeightProviderService`, `TerrainSurfaceProviderService`, etc) in `GetProvidedServices`.
+* Support thread safety.
+* Call `OnCompositionChanged` or `OnCompositionRegionChanged` whenever any data within the component changes.
+* Call the appropriate APIs on the `TerrainSystemServiceRequestBus` to register/unregister/refresh a terrain region.
+
+## Terrain physics integration
+
+The terrain physics support is provided through a divided component system. One half is the `TerrainPhysicsCollider`, which knows about the generic `TerrainDataRequestBus` for terrain data and the generic `HeightfieldProvider` buses for physics communication, and the other half is the `PhysXHeightfieldCollider`, which knows about PhysX and heightfields, but not about terrain. To integrate terrain with a new physics system, a new `HeightfieldCollider` component would need to be implemented to replace the PhysX version that works with the `HeightfieldProvider` buses and with the new physics system.
+
+## Terrain renderer
+
+The entire terrain renderer could be replaced by creating new rendering level components, rendering entity components, feature processors, and shaders. Replacing the renderer will have no effect on the base terrain system or the terrain physics integrations. This undertaking isn't for the faint of heart, as a high-performance terrain renderer can be difficult to write.
```

Comment:
I don't think the last sentence is well suited for a developer guide style (perhaps a blog style). While it is true, it may come off as discouraging to some. 
```suggestion
The entire terrain renderer can be replaced by creating new rendering level components, rendering entity components, feature processors, and shaders. Replacing the renderer doesn't affect the base terrain system or the terrain physics integrations.
```